tests/checks/mock/test_fail2ban.py

```diff
@@ -0,0 +1,167 @@
+                "currently_banned": "2",
+                "ip_list": "104.217.154.54 222.186.56.43",
+                "total_banned": "4985"
+            }
+        }
+        exptected = [
+            ("ssh", "fail2ban.action.currently_banned", "2"),
+            ("ssh", "fail2ban.action.total_banned", "4985"),
+            ("ssh", "fail2ban.filter.currently_failed", "2"),
+            ("ssh", "fail2ban.filter.total_failed", "62219")
+        ]
+        check = self._get_check()
+        with mock.patch.object(check, "get_jail_status") as get_jail_status:
+            with mock.patch.object(check, "get_jails") as get_jails:
+                get_jails.return_value = jails
+                get_jail_status.return_value = jail_status
+                stats = check.get_jail_stats()
+                self.assertEquals(exptected, list(stats))
+
+    def test_can_ping_fail2ban_pong(self):
+        check = self._get_check()
+        with mock.patch.object(check, "execute_command") as execute_command:
+            execute_command.return_value = ["Server replied: pong"]
+            self.assertTrue(check.can_ping_fail2ban())
+            execute_command.assert_called_with(["fail2ban-client", "ping"], sudo=False)
+
+    def test_can_ping_fail2ban_fail(self):
+        check = self._get_check()
+        with mock.patch.object(check, "execute_command") as execute_command:
+            # if it cannot connect we will get a subprocess.CalledProcessError
+            # which means execute_command will return []
+            execute_command.return_value = []
+            self.assertFalse(check.can_ping_fail2ban())
+            execute_command.assert_called_with(["fail2ban-client", "ping"], sudo=False)
+
+    def test_check(self):
+        jail_stats = [
+            ("ssh", "fail2ban.action.currently_banned", "2"),
+            ("ssh", "fail2ban.action.total_banned", "4985"),
+            ("ssh", "fail2ban.filter.currently_failed", "2"),
+            ("ssh", "fail2ban.filter.total_failed", "62219")
+        ]
+
+        expected_metrics = [
+            ('fail2ban.filter.total_failed', '62219', {'type': 'gauge', 'tags': ['jail:ssh']}),
+            ('fail2ban.action.total_banned', '4985', {'type': 'gauge', 'tags': ['jail:ssh']}),
+            ('fail2ban.action.currently_banned', '2', {'type': 'gauge', 'tags': ['jail:ssh']}),
+            ('fail2ban.filter.currently_failed', '2', {'type': 'gauge', 'tags': ['jail:ssh']})
+        ]
+
+        check, instances = get_check("fail2ban", self.config)
+        with mock.patch.object(check, "can_ping_fail2ban") as can_ping_fail2ban:
+            with mock.patch.object(check, "get_jail_stats") as get_jail_stats:
+                can_ping_fail2ban.return_value = True
+                get_jail_stats.return_value = jail_stats
+
+                check.check(instances[0])
+                self._assert_expected_metrics(expected_metrics, check.get_metrics())
+
+    def _assert_expected_metrics(self, expected, actual):
```

Comment:
[AgentCheckTest.assertMetrics](https://github.com/DataDog/dd-agent/blob/745d0ccd4ae8d9bcf4aa87540e552f8ae744c275/tests/checks/common.py#L293) will save you some trouble here.